src/business/runtime-state/collection/mutable-facade-map.ts

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import {type FacadeMap} from './facade-map.js';
+import {type Facade} from '../facade/facade.js';
+import {type FacadeObjectConstructor} from '../facade/facade-object-constructor.js';
+import {type ClassConstructor} from '../../utils/class-constructor.type.js';
+
+export class MutableFacadeMap<K, V extends Facade<BV>, BV> implements FacadeMap<K, V, BV> {
```

Comment:
We should see if we can eliminate the value constraint on Facade.
```suggestion
export class MutableFacadeMap<K, V> implements FacadeMap<K, V> {
```

Reply:
created https://github.com/hiero-ledger/solo/issues/1975 to resolve this in to not hold this PR up.